torchgeo/datamodules/deepglobelandcover.py

```diff
@@ -23,26 +29,45 @@ class DeepGlobeLandCoverDataModule(pl.LightningDataModule):
 
     def __init__(
         self,
-        batch_size: int = 64,
+        train_batch_size: int = 32,
         num_workers: int = 0,
+        patch_size: Union[Tuple[int, int], int] = (64, 64),
+        num_patches_per_tile: int = 32,
         val_split_pct: float = 0.2,
         **kwargs: Any,
     ) -> None:
-        """Initialize a LightningDataModule for DeepGlobe Land Cover based DataLoaders.
+        """Initialize a LightningDataModule for DeepGlobeLandCover based DataLoaders.
 
         Args:
-            batch_size: The batch size to use in all created DataLoaders
+            train_batch_size: The batch size used in the train DataLoader
```

Comment:
I think this should be named "number of tiles per batch", if I set "train_batch_size" to be 32, I will expect the batch size to be 32, not 32*`num_patches_per_tile`

Reply:
@nilsleh, thoughts on this?

Reply:
If you rename it to `num_tiles_per_batch` , do you intend that the effective batch_size would be `num_patches_per_tile` * `num_tiles_per_batch`? Then we can reverse the doc string a bit to explain that.

Reply:
Actually, I think setting `batch_size` and `num_tiles_per_batch` explicitly (as users might expect to be able to set a batch size) then calculating `num_patches_per_tiles = batch_size // num_tiles_per_batch` makes the most sense.

Reply:
That makes sense. However, then the `batch_size` you are setting is also not guaranteed to end up the same as what you specify, if you have `batch_size=16` and `num_tiles_per_batch=3` for example, then you would end up with a batch_size of just 15 at the end, so maybe suggest that it should be evenly divisible in the docstring?

Reply:
Very good points (thank you for taking the time to think through this)!

I think suggesting that `num_tiles_per_batch` should evenly divide `batch_size` (and warning if not) is a reasonable way to go so that we can keep `batch_size` as an argument.

> Also you would have to check or warn that a tile actually has enough patches. For example, if batch_size=32 and num_tiles_per_batch=1 ...

I'm not too worried about this as we can set reasonable defaults. Anyone who changes them should be looking at tile sizes anyway. (e.g. as a user, I would only imagine setting `num_tiles_per_batch=1` if I was trying something particularly crazy, because generally you would want batches with diverse sets of patches in them).

Reply:
Yes, that is reasonable, I made a commit accordingly with a "draft" for a warning message.

Reply:
Good discussion, my only comment is that we should be careful of the use of the word "tile" since it can mean different things to different people: #516. Maybe "patch" or "chip" instead?

Reply:
in this case the name `num_tiles_per_batch` is quiet accurate, right? Because it specifies from how many tiles/scenes in the dataset to take random cropped patches from to overall yield an overall `batch_size`. Or based on #516, would `num_scenes_per_batch` be better?

Reply:
What I'm saying is that "tile" and "scene" mean different things to different people, so we should avoid using them. I think `num_patches_per_batch` or `num_chips_per_batch` may be better.

Reply:
Oh wait, it's not patches/chips, it's files. We may want to make a decision on #516 before we decide on a name for this one.

Reply:
I think "tiles" is fine -- we can change later if needed but I don't want to hold up this

Reply:
I think I actually want to reverse the decision made here. Allowing users to specify `batch_size` and `num_tiles_per_batch` introduces a lot of error checking (`batch_size` must be > `num_tiles_per_batch`) and adjustments/warning (when `batch_size` is not a multiple of `num_tiles_per_batch`). I prefer @nilsleh's original suggestion of `num_tiles_per_batch` and `num_patches_per_tile`.